src/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -170,11 +173,6 @@ app.directive('dashboardApp', function ($injector) {
       };
 
       $scope.updateQueryAndFetch = function (query) {
-        // reset state if language changes
-        if ($scope.model.query.language && $scope.model.query.language !== query.language) {
```

Comment:
Is this safe to remove? Since I'm still getting familiar with how this all works, can you explain why?

Reply:
Yeah, previously when kuery was enabled we removed the filter bar and put everything in the query bar. So when switching languages we wanted to clear out the filters in app state. Now the filter bar is present for both lucene and kuery, so there's no need to blow away the users filters when switching languages. This change really should have been made in the PR that brought back the filter bar for kuery but I only noticed the issue while working on this PR.